### PR TITLE
IGNITE-14474 expand log for errors with GridDhtPartitionSupplyMessage in GridDhtPartitionDemander

### PR DESCRIPTION
**Pull request by RodionSmolnikovGG**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java [hunk 1]
```diff
@@ -536,22 +538,28 @@ else if (!rebalanceFut.isActual(supplyMsg.rebalanceId()))
                 log.debug("Received supply message [" + demandRoutineInfo(nodeId, supplyMsg) + ']');
 
             // Check whether there were error during supplying process.
+            Throwable msgExc = null;
+            final GridDhtPartitionTopology top = grp.topology();
```
Comment (denis-chudov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:542]: please add empty line

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java [hunk 2]
```diff
@@ -536,22 +538,28 @@ else if (!rebalanceFut.isActual(supplyMsg.rebalanceId()))
...
             else if (supplyMsg.error() != null)
-                errMsg = supplyMsg.error().getMessage();
+                msgExc = supplyMsg.error();
 
-            if (errMsg != null) {
-                U.warn(log, "Rebalancing routine has failed [" +
-                    demandRoutineInfo(nodeId, supplyMsg) + ", err=" + errMsg + ']');
+            if (msgExc != null) {
+                GridDhtPartitionMap partMap = top.localPartitionMap();
+                Set<Integer> unstableParts = supplyMsg.infos().keySet().stream()
```
Comment (denis-chudov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:551]: please add empty line

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java [hunk 3]
```diff
@@ -536,22 +538,28 @@ else if (!rebalanceFut.isActual(supplyMsg.rebalanceId()))
...
+                    .filter(p -> partMap.get(p) == MOVING)
+                    .collect(Collectors.toSet());
+
+                U.error(log, "Rebalancing routine has failed [" + demandRoutineInfo(nodeId, supplyMsg) + "]" +
+                    ", Partitions could be unavailable for reading " + S.compact(unstableParts), msgExc);
```
Comment (denis-chudov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:556]: According to Ignite code style, such messages should look like `Message text [key0=value0, ..., keyN=valueN]` [1]
I suggest to change message to `Rebalancing routine has failed, some partitions could be unavailable for reading [..., unavailablePartitions=[...]]`

[1] https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-StringOutput

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java (hunk 1 above)
Comment (RodionSmolnikovGG) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:542]: done

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java (hunk 2 above)
Comment (RodionSmolnikovGG) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:551]: done

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java (hunk 3 above)
Comment (RodionSmolnikovGG) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java]: Added log listener

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/rebalancing/GridCacheRebalancingUnmarshallingFailedSelfTest.java [hunk 4]
```diff
@@ -176,6 +192,8 @@ private void runTest() throws Exception {
 
         for (int i = 50; i < 100; i++)
             assertNull(grid(1).cache(CACHE).get(new TestKey(String.valueOf(i))));
+
+        assertTrue("Unmarshal log error messaage is not valid.", unmarshalErrorLogListener.check());
```
Comment (denis-chudov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/rebalancing/GridCacheRebalancingUnmarshallingFailedSelfTest.java:196]: typo - "messaage"

**Review by denis-chudov — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java [hunk 5]
```diff
@@ -536,22 +537,31 @@ else if (!rebalanceFut.isActual(supplyMsg.rebalanceId()))
...
+            if (msgExc != null) {
+                GridDhtPartitionMap partMap = top.localPartitionMap();
+
+                Set<Integer> unstableParts = supplyMsg.infos().keySet().stream()
+                    .filter(p -> partMap.get(p) == MOVING)
+                    .collect(Collectors.toSet());
+
+                U.error(log, "Rebalancing routine has failed, some partitions could be unavailable for reading" +
+                    " [" + demandRoutineInfo(nodeId, supplyMsg) +
+                    ", unavailablePartitions=" + S.compact(unstableParts) + "]", msgExc);
```
Comment (sk0x50) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java:558]: Please change "]" to \']\'

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/rebalancing/GridCacheRebalancingUnmarshallingFailedSelfTest.java [hunk 6]
```diff
@@ -41,14 +44,21 @@
...
     protected static AtomicInteger readCnt = new AtomicInteger();
 
     /** */
     private volatile Marshaller marshaller;
 
+    /** */
+    private ListeningTestLogger customLog;
+
+    /** */
+    private static final Pattern unmarshalingErrorPattern = Pattern.compile(".*Rebalancing routine has failed" +
```
[the remark this answers is not in the document]
Comment (RodionSmolnikovGG) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/rebalancing/GridCacheRebalancingUnmarshallingFailedSelfTest.java]: Done

File: modules/ssh/src/main/java/org/apache/ignite/internal/util/nodestart/StartNodeCallableImpl.java [hunk 7]
```diff
@@ -203,7 +203,7 @@ public StartNodeCallableImpl(IgniteRemoteStartSpecification spec, int timeout) {
 
             shell(ses, "mkdir " + scriptOutputDir);
 
-            String scriptOutputPath = scriptOutputDir + separator + scriptOutputFileName;
+            String scriptOutputPath = "C:\\Users\\user\\log.log" + separator + scriptOutputFileName;
```
Comment (alievmirza) [inline modules/ssh/src/main/java/org/apache/ignite/internal/util/nodestart/StartNodeCallableImpl.java:206]: ?

**Review by alievmirza — CHANGES_REQUESTED** [review]

File: modules/ssh/src/main/java/org/apache/ignite/internal/util/nodestart/StartNodeCallableImpl.java (hunk 7 above)
Comment (RodionSmolnikovGG) [reply modules/ssh/src/main/java/org/apache/ignite/internal/util/nodestart/StartNodeCallableImpl.java]: something went wrong - part of dev commit of other ticket - reverted

**Pull request closed without merging**